src/pipecat/services/riva.py

```diff
@@ -0,0 +1,264 @@
+    logger.error(f"Exception: {e}")
+    logger.error(
+        "In order to use nvidia riva TTS or STT, you need to `pip install pipecat-ai[riva]`. Also, set `NVIDIA_API_KEY` environment variable."
+    )
+    raise Exception(f"Missing module: {e}")
+
+
+class FastpitchTTSService(TTSService):
+    class InputParams(BaseModel):
+        language: Optional[Language] = Language.EN_US
+
+    def __init__(
+        self,
+        *,
+        api_key: str,
+        server: str = "grpc.nvcf.nvidia.com:443",
+        voice_id: str = "English-US.Female-1",
+        sample_rate_hz: int = 24000,
+        # nvidia riva calls this 'function-id'
+        model: str = "0149dedb-2be8-4195-b9a0-e57e0e14f972",
+        params: InputParams = InputParams(),
+        quality: int = 20,
+        **kwargs,
+    ):
+        super().__init__(sample_rate=sample_rate_hz, **kwargs)
+        self._api_key = api_key
+        self._voice_id = voice_id
+        self._sample_rate_hz = sample_rate_hz
+        self._language_code = params.language
+        self.quality = quality
+
+        self.set_model_name("fastpitch-hifigan-tts")
+        self.set_voice(voice_id)
+
+        metadata = [
+            ["function-id", model],
+            ["authorization", f"Bearer {api_key}"],
+        ]
+        auth = riva.client.Auth(None, True, server, metadata)
+
+        self.service = riva.client.SpeechSynthesisService(auth)
+
+    async def run_tts(self, text: str) -> AsyncGenerator[Frame, None]:
+        def read_audio_responses():
+            try:
+                responses = self.service.synthesize_online(
+                    text,
+                    self._voice_id,
+                    self._language_code,
+                    sample_rate_hz=self._sample_rate_hz,
+                    audio_prompt_file=None,
+                    quality=self.quality,
+                    custom_dictionary={},
+                )
+                return responses
+            except Exception as e:
+                logger.error(f"{self} exception: {e}")
+                return []
+
+        logger.debug(f"Generating TTS: [{text}]")
+
+        await self.start_ttfb_metrics()
+        yield TTSStartedFrame()
+
+        responses = await asyncio.to_thread(read_audio_responses)
+
+        for resp in responses:
+            await self.stop_ttfb_metrics()
+
+            frame = TTSAudioRawFrame(
+                audio=resp.audio,
+                sample_rate=self._sample_rate_hz,
+                num_channels=1,
+            )
+            yield frame
+
+        await self.start_tts_usage_metrics(text)
+        yield TTSStoppedFrame()
+
+
+class ParakeetSTTService(STTService):
+    class InputParams(BaseModel):
+        language: Optional[Language] = Language.EN_US
+
+    def __init__(
+        self,
+        *,
+        api_key: str,
+        server: str = "grpc.nvcf.nvidia.com:443",
+        # nvidia calls this 'function-id'
+        model: str = "1598d209-5e27-4d3c-8079-4751568b1081",
```

Comment:
what is this UUID? is it specific to our account?

Reply:
It is not specific to our account.  It is the parakeet / riva ASR function id
https://build.nvidia.com/nvidia/parakeet-ctc-1_1b-asr/api
We could just remove "model" and call this function_id.

Reply:
Oh, then yes, I think we should call this `function_id`.